lib/parsers/yarn2-lock-parser.ts

```diff
@@ -81,21 +90,36 @@ export class Yarn2LockParser extends LockParserBase {
     return depTreeWithMeta;
   }
 
-  protected getDepMap(lockfile: Lockfile): DepMap {
+  protected getDepMap(
+    lockfile: Lockfile,
+    resolutions?: ManifestDependencies,
+  ): DepMap {
     const yarnLockfile = lockfile as Yarn2Lock;
     const depMap: DepMap = {};
 
+    const dependencies = lockfile.dependencies || {};
```

Comment:
Add `YarnLockDeps` type (instead of casting later)

Reply:
Cast is still necessary as a type annotation on variable to powerful to cast from this type for whatever reason: `dependencies: PackageLockDeps | YarnLockDeps`

Reply:
Moved cast to this line though to make it more sensible